From 633c7a051fee5a58740c4701e3e49098b9a17e28 Mon Sep 17 00:00:00 2001
From: vladimir.a.ivanov <vladimir.a.ivanov@intel.com>
Date: Tue, 10 Dec 2013 15:43:30 +0700
Subject: Dalvik: non vectorized loops should be rejected by vectorizer

BZ: 157338

If no expressions detected in the loop for vectorization this loop
is vectorized. The missed check was added.

Change-Id: Ibba1da9858210db58f6d742bd8e04e1793a80fb2
Category: device enablement
Domain: AOSP.Dalvik-Other
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: vladimir.a.ivanov <vladimir.a.ivanov@intel.com>
---
 vm/compiler/Vectorization.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/vm/compiler/Vectorization.cpp b/vm/compiler/Vectorization.cpp
index 31c3b6a..1064199 100644
--- a/vm/compiler/Vectorization.cpp
+++ b/vm/compiler/Vectorization.cpp
@@ -1397,6 +1397,11 @@ static bool haveSafeAccumulation (CompilationUnit *cUnit, LoopInformation *loopI
     std::vector<Expression *> ivExpressions;
     dvmCompilerGetLoopExpressions (cUnit, loopInfo, ivExpressions);
 
+    // if no expressions nothing to process
+    if (ivExpressions.size() == 0) {
+        return false;
+    }
+
     for (std::vector<Expression *>::iterator it = ivExpressions.begin(); it != ivExpressions.end(); it++)
     {
         BytecodeExpression *bcExpression = static_cast<BytecodeExpression *>(*it);
-- 
1.7.4.1

